src/index.ts

```diff
@@ -35,68 +33,72 @@ const collectImageUrls = (jsonObject: JsonObject | string, resultArray: string[]
     }
 }
 
-// return json object from json file
-const getJsonObjectFromJsonFile = (filePath: string): JsonObject => JSON.parse(fs.readFileSync(filePath).toString())
-
 // create empty json file
-const createJsonFile = (filePath: string) => {
+const createJsonFile = (filePath: string, jsonObject: JsonObject) => {
     const directoryName = path.dirname(filePath)
     // create directory if not exist
     if (!fs.existsSync(directoryName)) {
         fs.mkdirSync(directoryName, { recursive: true })
 
     }
     // create empty json file
-    fs.writeFileSync(filePath, JSON.stringify({}))
+    fs.writeFileSync(filePath, JSON.stringify(jsonObject, null, 2))
 }
 
-// append key value pair to json file
-const appendKeyValueToJsonFile = (filePath: string, key: string, value: string) => {
-    const json = JSON.parse(fs.readFileSync(filePath).toString())
-    json[key] = value;
-    fs.writeFileSync(filePath, JSON.stringify(json))
+// remove all files in directory
+const removeFilesInDirectory = (dirPath: string) => {
+    const files = fs.readdirSync(dirPath);
+    for (const file of files) {
+        fs.unlinkSync(path.join(dirPath, file))
+    }
 }
 
 const main = async () => {
     // get user input
     program.parse(process.argv)
     const url = program.url
-    const imageOutputDirFilepath = program.output
-    const cacheFilepath = program.cache
+    const imageOutputDirFilepath =  './generated/images'
+    const mapFilePath =  './generated/map.json'
```

Comment:
Personally I think having an option to input file name but with these as defaults is better than simply forcing the file to take this path. Not major though.